crates/yarn_slinger/tests/dialogue_tests.rs

```diff
@@ -188,94 +191,67 @@ fn test_function_argument_type_inference() {
-            let expected_step = test_plan
-                .read()
-                .unwrap()
-                .as_ref()
-                .unwrap()
-                .next_expected_step;
-            assert_eq!(ExpectedStepType::Stop, expected_step);
-        });
-
     let result = Compiler::from_test_source("-> option 1\n->option 2\nfinal line\n")
         .compile()
         .unwrap();
 
-    test_base.with_compilation(result).run_standard_testcase();
+    let mut test_base = TestBase::new()
+        .with_test_plan(
+            TestPlan::new()
+                .expect_option("option 1")
+                .expect_option("option 2")
+                .then_select(0)
+                .expect_line("final line"),
+        )
+        .with_compilation(result);
+    test_base.dialogue.set_node_to_start();
+
+    while let Some(events) = test_base.dialogue.continue_() {
+        for event in events {
+            match event {
+                DialogueEvent::Line(line) => {
+                    let line_text = test_base.string_table.get(&line.id).unwrap().text.clone();
+                    let parsed_text = test_base.dialogue.parse_markup(&line_text);
+                    let test_plan = test_base.test_plan.as_mut().unwrap();
+                    test_plan.next();
+
+                    let expected_step = test_plan.next_expected_step;
+                    let expected_value = test_plan.next_step_value.clone().unwrap();
+                    assert_eq!(ExpectedStepType::Line, expected_step);
+                    assert_eq!(StepValue::String(parsed_text), expected_value);
+                }
+                DialogueEvent::Options(options) => {
+                    test_base.test_plan.as_mut().unwrap().next();
+                    // Assert that the list of options we were given is
+                    // identical to the list of options we expect
+                    let actual_options: Vec<_> = options
+                        .into_iter()
+                        .map(|o| ProcessedOption {
+                            line: test_base.get_composed_text_for_line(&o.line),
+                            enabled: o.is_available,
+                        })
+                        .collect();
+                    let test_plan = test_base.test_plan.as_ref().unwrap();
+                    let next_expected_options = test_plan.next_expected_options.clone();
+                    assert_eq!(next_expected_options, actual_options);
+
+                    let expected_step = test_plan.next_expected_step;
+                    assert_eq!(ExpectedStepType::Select, expected_step);
+                    test_base
+                        .dialogue
+                        .set_selected_option(OptionId::construct_for_debugging(0));
+                }
+                DialogueEvent::Command(_) => {}
+                DialogueEvent::NodeComplete(_) => {}
+                DialogueEvent::NodeStart(_) => {}
+                DialogueEvent::PrepareForLines(_) => {}
```

Comment:
Since these all match to the same thing, there are two ways to simplify this:
Either we use `|` to concatenate these options to match any of them, or we use the `_` wildcard at the end of the match.

Reply:
I prefer the wildcard, as rust has compile time checking on matches, right?

Reply:
[`c83a56c` (#103)](https://github.com/yarn-slinger/yarn-slinger/pull/103/commits/c83a56ceaea10bd361d56659d0fcf3d5b309ef2a)